src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -54,16 +54,23 @@ public DefaultJedisSocketFactory(HostAndPort hostAndPort, JedisClientConfig conf
     }
   }
 
-  private void connectToFirstSuccsefulHost(Socket socket, HostAndPort hostAndPort) throws Exception {
+  private Socket connectToFirstSuccsefulHost(HostAndPort hostAndPort) throws Exception {
     List<InetAddress> hosts = Arrays.asList(InetAddress.getAllByName(hostAndPort.getHost()));
     if (hosts.size() > 1) {
       Collections.shuffle(hosts);
     }
+
     JedisConnectionException jce = new JedisConnectionException("Failed to connect to any host resolved for DNS name.");
     for (InetAddress host : hosts) {
+      Socket socket = new Socket();
+      socket.setReuseAddress(true);
+      socket.setKeepAlive(true); // Will monitor the TCP connection is valid
+      socket.setTcpNoDelay(true); // Socket buffer Whetherclosed, to ensure timely delivery of data
+      socket.setSoLinger(true, 0); // Control calls close () method, the underlying socket is closed immediately
+      socket.setSoTimeout(socketTimeout);
```

Comment:
I understand this won't make much difference but IMO we should move this block within following `try` which would then be more similar to previous version of code.

Reply:
Agree, already updated.